### ISSUE TITLE
Source type vector not recognized as a metric source

### ISSUE THREAD

**Issue by sgowie** [timberio/vector#1153]
According to the documentation, the `vector` source should provide both log and metrics

> The vector source ingests data through another upstream Vector instance and outputs log and metric events.

However, a simple source to sink flow refuses to start due to a data type mismatch.

**Example Config**

```toml
data_dir = "/var/lib/vector"

[sources.debug]
type = "vector"
address = "0.0.0.0:9000"


[sinks.prometheus]
type = "prometheus"
inputs = ["debug"]
namespace = "test"
buckets = [0.0, 10.0, 100.0, 1000.0, 10000.0, 100001.0]
address = "0.0.0.0:9598"
```

**Invocation:**

`docker run -p 9000:9000 -p 9598:9598 -v $(pwd):/etc/vector -v $(pwd)/data:/var/lib/vector timberio/vector:latest-alpine --config /etc/vector/vector.toml`

**Result:**

```
Nov 05 22:48:50.226 INFO vector: Log level "info" is enabled.
Nov 05 22:48:50.230 INFO vector: Loading config. path="/etc/vector/vector.toml"
Nov 05 22:48:50.233 INFO vector: Vector is starting. version="0.4.0-dev" git_version="v0.5.0" released="Thu, 10 Oct 2019 18:41:05 GMT" arch="x86_64"
Nov 05 22:48:50.233 ERROR vector::topology: Configuration error: Data type mismatch between debug (Log) and prometheus (Metric)
```

**Comment by lukesteensen** [comment]
Sorry about that @sgowie! That would be a bug in our documentation currently, though we do plan to support this.

We have issue #421 to track the actual implementation work that needs to happen before we can support a source that outputs both logs and metrics.

**Comment by a-rodin** [comment]
Closed by #1153.
